test/Database/LSMTree/Model/Session.hs

```diff
@@ -246,59 +250,60 @@ runModelMWithInjectedErrors ::
 runModelMWithInjectedErrors Nothing onNoErrors _ st =
     runModelM onNoErrors st
 runModelMWithInjectedErrors (Just _) _ onErrors st =
-    runModelM (onErrors >> throwError DefaultErrDiskFault) st
+    runModelM (onErrors >> throwError ModelErrDiskFault) st
 
 -- | The default 'ErrDiskFault' that model operations will throw.
-pattern DefaultErrDiskFault :: Err
-pattern DefaultErrDiskFault = ErrDiskFault "default"
+pattern ModelErrDiskFault :: Err
+pattern ModelErrDiskFault = ErrDiskFault "model does not produce an error message"
 
--- | The default 'ErrSnapshotCorrupted' that model operations will throw.
-pattern DefaultErrSnapshotCorrupted :: Err
-pattern DefaultErrSnapshotCorrupted = ErrSnapshotCorrupted "default"
-
---
+-----
 -- Errors
 --
 
-data Err =
-    ErrTableClosed
-  | ErrSnapshotExists
-  | ErrSnapshotDoesNotExist
-  | ErrSnapshotWrongType
-  | ErrBlobRefInvalidated
+data Err
```

Comment:
I wonder if we can leave out the errors that the model never throws, like `ErrTableTypeMismatch`. If we don't model certain behaviour, then maybe we don't need to represent the errors. Then, SUT errors that have no model equivalent can be mapped using `ErrOther . show`

Reply:
I'm taking `ErrOther` to only be used for errors not thrown by the library API, so that it's easier to see if something unusual is going on.

Reply:
Let's leave `ErrTableTypeMismatch` for a future PR to sort out, because it looks like the _real_ implementation doesn't throw it either, so that might require some input from the folks working on unions.